dc/s2n-quic-dc/src/event/generated.rs

```diff
@@ -0,0 +1,686 @@
+        pub frame: Frame,
+    }
+    impl Event for FrameSent {
+        const NAME: &'static str = "transport:frame_sent";
+    }
+}
+#[cfg(feature = "event-tracing")]
+pub mod tracing {
+    #![doc = r" This module contains event integration with [`tracing`](https://docs.rs/tracing)"]
+    use super::api;
+    #[doc = r" Emits events with [`tracing`](https://docs.rs/tracing)"]
+    #[derive(Clone, Debug)]
+    pub struct Subscriber {
+        client: tracing::Span,
+        server: tracing::Span,
+    }
+    impl Default for Subscriber {
+        fn default() -> Self {
+            let root =
+                tracing :: span ! (target : "s2n_quic" , tracing :: Level :: DEBUG , "s2n_quic");
+            let client =
+                tracing :: span ! (parent : root . id () , tracing :: Level :: DEBUG , "client");
+            let server =
+                tracing :: span ! (parent : root . id () , tracing :: Level :: DEBUG , "server");
+            Self { client, server }
+        }
+    }
+    impl super::Subscriber for Subscriber {
+        type ConnectionContext = tracing::Span;
+        fn create_connection_context(
+            &self,
+            meta: &api::ConnectionMeta,
+            _info: &api::ConnectionInfo,
+        ) -> Self::ConnectionContext {
+            let parent = match meta.endpoint_type {
+                api::EndpointType::Client {} => self.client.id(),
+                api::EndpointType::Server {} => self.server.id(),
+            };
+            tracing :: span ! (target : "s2n_quic" , parent : parent , tracing :: Level :: DEBUG , "conn" , id = meta . id)
+        }
+        #[inline]
+        fn on_frame_sent(
+            &self,
+            context: &Self::ConnectionContext,
+            _meta: &api::ConnectionMeta,
+            event: &api::FrameSent,
+        ) {
+            let id = context.id();
+            let api::FrameSent {
+                packet_header,
+                path_id,
+                frame,
+            } = event;
+            tracing :: event ! (target : "frame_sent" , parent : id , tracing :: Level :: DEBUG , packet_header = tracing :: field :: debug (packet_header) , path_id = tracing :: field :: debug (path_id) , frame = tracing :: field :: debug (frame));
+        }
+    }
+}
+pub mod builder {
+    use super::*;
+    #[derive(Clone, Debug)]
+    #[doc = " Frame was sent"]
+    pub struct FrameSent {
+        pub packet_header: PacketHeader,
+        pub path_id: u64,
+        pub frame: Frame,
+    }
+    impl IntoEvent<api::FrameSent> for FrameSent {
+        #[inline]
+        fn into_event(self) -> api::FrameSent {
+            let FrameSent {
+                packet_header,
+                path_id,
+                frame,
+            } = self;
+            api::FrameSent {
+                packet_header: packet_header.into_event(),
+                path_id: path_id.into_event(),
+                frame: frame.into_event(),
+            }
+        }
+    }
+}
+pub mod supervisor {
```

Comment:
I don't think supervisors will be supported with this initial pass so let's probably omit it for now